kratos/input_output/cad_json_input.h

```diff
@@ -0,0 +1,823 @@
+                << "For BrepSurface \"" << GetIdOrName(rParameters) << "\""
+                << "\", is_trimmed is not provided in the input."
+                << " is_trimmed = true is considered." << std::endl;
+        }
+
+        if (rParameters.Has("boundary_loops"))
+        {
+            BrepCurveOnSurfaceLoopArrayType outer_loops, inner_loops;
+            tie(outer_loops, inner_loops) =
+                ReadBoundaryLoops(rParameters["boundary_loops"], p_surface, rModelPart, EchoLevel);
+
+            auto p_brep_surface =
+                Kratos::make_shared<BrepSurfaceType>(
+                    p_surface,
+                    outer_loops,
+                    inner_loops,
+                    is_trimmed);
+
+            SetIdOrName<BrepSurfaceType>(rParameters, p_brep_surface);
+
+            rModelPart.AddGeometry(p_brep_surface);
+        }
+        else
+        {
+            KRATOS_INFO_IF("ReadBrepSurface", (EchoLevel > 4))
+                << "For BrepSurface \"" << GetIdOrName(rParameters) << "\""
+                << "\", boundary_loops are not provided in the input."
+                << " It will be considered as untrimmed." << std::endl;
+
+            auto p_brep_surface =
+                Kratos::make_shared<BrepSurfaceType>(
+                    p_surface);
+
+            SetIdOrName<BrepSurfaceType>(rParameters, p_brep_surface);
+
+            rModelPart.AddGeometry(p_brep_surface);
+        }
+    }
+
+
+    ///@}
+    ///@name Read in Surface Trimming
+    ///@{
+
+    static BrepCurveOnSurfaceLoopType
+        ReadTrimmingCurveVector(
+            const Parameters& rParameters,
+            typename NurbsSurfaceType::Pointer pNurbsSurface,
+            ModelPart& rModelPart,
+            SizeType EchoLevel = 0)
+    {
+        KRATOS_ERROR_IF(rParameters.size() < 1)
+            << "Trimming curve list has no element." << std::endl;
+
+        BrepCurveOnSurfaceLoopType
+            trimming_brep_curve_vector(rParameters.size());
+
+        for (IndexType tc_idx = 0; tc_idx < rParameters.size(); tc_idx++)
+        {
+            trimming_brep_curve_vector[tc_idx] = ReadTrimmingCurve(
+                rParameters[tc_idx], pNurbsSurface, rModelPart, EchoLevel);
+        }
+
+        return trimming_brep_curve_vector;
+    }
+
+    static typename BrepCurveOnSurfaceType::Pointer
+        ReadTrimmingCurve(
+            const Parameters& rParameters,
+            typename NurbsSurfaceType::Pointer pNurbsSurface,
+            ModelPart& rModelPart,
+            SizeType EchoLevel = 0)
+    {
+        KRATOS_ERROR_IF_NOT(rParameters.Has("curve_direction"))
+            << "Missing 'curve_direction' in nurbs curve" << std::endl;
+        bool curve_direction = rParameters["curve_direction"].GetBool();
+
+        KRATOS_ERROR_IF_NOT(rParameters.Has("parameter_curve"))
+            << "Missing 'parameter_curve' in nurbs curve" << std::endl;
+
+        auto p_trimming_curve = ReadNurbsCurve<2, TEmbeddedNodeType>(

```

Comment:
```suggestion
        auto p_trimming_curve(ReadNurbsCurve<2, TEmbeddedNodeType>(
```

also applies more often here

Reply:
Changed it. But are you sure that this is more efficient?

Reply:
well in one case you construct the object directly
In other cases you additionally call the copy-ctor
(@pooyan-dadvand  please correct me if wrong)
I am too lazy to dig up a good source but you can take a look e.g. here: https://stackoverflow.com/questions/2462773/c-copy-construct-construct-and-assign-question